libs/api-test-environment/src/main/java/de/cyface/apitestutils/fixture/GeoLocationTestFixture.java

```diff
@@ -51,39 +57,47 @@ public final class GeoLocationTestFixture implements TestFixture {
      */
     public static final String TEST_USER_NAME = "admin";
     /**
-     * The measurements used during the test
+     * The identifiers of the measurements to be used during the test.
      */
-    private final List<TestMeasurementDocument> testMeasurementDocuments;
+    private final List<MeasurementIdentifier> testMeasurementIdentifiers;
 
     /**
      * Creates a new completely initialized fixture for the test.
      *
-     * @param testMeasurementDocuments The measurements used during the test
+     * @param testMeasurementIdentifiers The identifiers of the measurements to be used during the test.
      */
     @SuppressWarnings("unused") // API
-    public GeoLocationTestFixture(final List<TestMeasurementDocument> testMeasurementDocuments) {
-        this.testMeasurementDocuments = testMeasurementDocuments;
+    public GeoLocationTestFixture(final List<MeasurementIdentifier> testMeasurementIdentifiers) {
+        this.testMeasurementIdentifiers = testMeasurementIdentifiers;
     }
 
     @Override
-    public Future<Void> insertTestData(MongoClient mongoClient) {
-        final Promise<Void> createAuthUserPromise = Promise.promise();
-        final Promise<Void> createTestUserPromise = Promise.promise();
-        final var futures = new ArrayList<Future>();
-        futures.add(createAuthUserPromise.future());
-        futures.add(createTestUserPromise.future());
-        testMeasurementDocuments.forEach(document -> {
-            final Promise<Void> promise = Promise.promise();
-            futures.add(promise.future());
-            document.insert(mongoClient, promise);
+    public Future<String> insertTestData(MongoClient mongoClient) {
+
+        final var manager = new DirectTestUser(TEST_USER_NAME, "secret",
+                TEST_GROUP + DatabaseConstants.GROUP_MANAGER_ROLE_SUFFIX);
+        final var groupUser = new DirectTestUser(TEST_GROUP_USER_USERNAME, "secret",
+                TEST_GROUP + DatabaseConstants.USER_GROUP_ROLE_SUFFIX);
+        final var managerInsert = manager.insert(mongoClient);
+        final var userInsert = groupUser.insert(mongoClient);
+
+        final Promise<String> promise = Promise.promise();
+        userInsert.onSuccess(userId -> {
+            final var testDocuments = testMeasurementIdentifiers.stream().map(
+                    id -> new TestMeasurementDocument(userId, id.getMeasurementIdentifier(), id.getDeviceIdentifier()))
+                    .collect(Collectors.toList());
+
+            // noinspection rawtypes
+            final var futures = testDocuments.stream().map(d -> (Future)d.insert(mongoClient))
+                    .collect(Collectors.toList());
+            futures.add(managerInsert);
+
+            final CompositeFuture composition = CompositeFuture.all(futures);
+            composition.onSuccess(succeeded -> promise.complete(userId));
+            composition.onFailure(promise::fail);
         });
```

Comment:
The Fixtures now need to wait until the test user is created in the database to use this user's `id` in `userId`.